### DISCUSSION TITLE
Add support for json marshal/unmarshal

### DISCUSSION THREAD

**Discussion by patrickmao1** [umbracle/go-eth-consensus#34] [Ideas]
For now the JSON returned by Beacon APIs can't get unmarshalled into the types directly, but JSON is the default Beacon API response format. The workload shouldn't be as enormous as writing the SSZ unmarshallers since only the basic types such as uint and hashes.

**Comment by ferranbt** [comment]
Is this still an issue? The library does have encoding and decoding for JSON types [here](https://github.com/umbracle/go-eth-consensus/blob/main/http/encoding.go). It is tested end-to-end with the beacon open api spec.
